src/escape_string.cpp

```diff
@@ -591,8 +595,25 @@ namespace {
 
 #elif TORRENT_USE_LOCALE
 
+namespace {
+
+	bool ends_with(string_view s, string_view suffix)
+	{
+		return s.size() >= suffix.size()
+			&& s.substr(s.size() - suffix.size()) == suffix;
+	}
+
+	bool need_conversion()
+	{
+		static bool const ret = ends_with(std::locale("").name(), ".UTF-8");
```

Comment:
I just recently discovered that this `std::locale` constructor may throw, which is likely to terminate the process. I will need to fix this in `RC_2_0`. I'm not sure how to fix it yet, since the default constructor seems to always give you the "C" locale.